File: Perl/scripts/Problem005.pl
sub gcd {
	my ($x, $y) = @_;
	while ($x) { ($x, $y) = ($y % $x, $x) }
	$y
}
 
sub lcm {
	my ($x, $y) = @_;
	($x && $y) and $x / gcd($x, $y) * $y or 0
}

$start = time();
$ans = 1;
for (my $i = 1; $i < 21; $i++){
    $ans = lcm($ans, $i+1);
}

$exc_time = time() - $start;
print "\nAnswer: $ans\n";
print "Time Taken: $exc_time seconds\n\n";

File: Python/scripts/Problem005.py
from time import time
from math import lcm
from functools import reduce


main = lambda: reduce(lcm, range(1, 21))


if __name__ == '__main__':
    start = time()
    print(f"\nAnswer: { main() }")
    print(f"Time Taken: { time() - start }\n")
